tools/registrar-scraper/src/index.ts

```diff
@@ -52,6 +52,67 @@ async function main() {
       data: Object.values(instructorInfo),
     }),
   ]);
+  const courses = Object.fromEntries((await prisma.course.findMany()).map((x) => [x.id, x]));
+  const instructors = Object.fromEntries(
+    (await prisma.instructor.findMany()).map((x) => [x.ucinetid, x]),
+  );
+  const newCourses = [];
+  for (const course of Object.values(courses)) {
+    newCourses.push({
+      ...course,
+      prerequisiteTree: course.prerequisiteTree as object,
+      instructors: course.instructorHistory
+        .map((x) => instructors[x])
+        .filter((x) => x)
+        .map(({ ucinetid, name, shortenedName }) => ({ ucinetid, name, shortenedName })),
+      prerequisites: course.prerequisiteList
+        .map((x) => courses[x.replace(/ /g, "")])
+        .filter((x) => x)
+        .map(({ id, title, department, courseNumber }) => ({
+          id,
+          title,
+          department,
+          courseNumber,
+        })),
+      dependencies: course.prerequisiteFor
+        .map((x) => courses[x.replace(/ /g, "")])
+        .filter((x) => x)
+        .map(({ id, title, department, courseNumber }) => ({
+          id,
+          title,
+          department,
+          courseNumber,
+        })),
+    });
+  }
```

Comment:
Functional programming 😎 
```suggestion
  const newCourses = Object.values(courses).map(course => {
    return {
      ...course,
      prerequisiteTree: course.prerequisiteTree as object,
      instructors: course.instructorHistory
        .map((x) => instructors[x])
        .filter((x) => x)
        .map(({ ucinetid, name, shortenedName }) => ({ ucinetid, name, shortenedName })),
      prerequisites: course.prerequisiteList
        .map((x) => courses[x.replace(/ /g, "")])
        .filter((x) => x)
        .map(({ id, title, department, courseNumber }) => ({
          id,
          title,
          department,
          courseNumber,
        })),
      dependencies: course.prerequisiteFor
        .map((x) => courses[x.replace(/ /g, "")])
        .filter((x) => x)
        .map(({ id, title, department, courseNumber }) => ({
          id,
          title,
          department,
          courseNumber,
        })),
    }
  })
```